File: examples/get_info.py
import rbpodo as rb

ROBOT_IP = "10.0.2.7"


def _main():
    try:
        robot = rb.Cobot(ROBOT_IP)
        rc = rb.ResponseCollector()

        print(robot.get_tcp_info(rc))

        print(robot.get_tfc_info(rc))
    except Exception as e:
        print(e)


if __name__ == "__main__":
    _main()

File: examples/get_info.cpp
#include <iostream>
#include <array>
#include "rbpodo/rbpodo.hpp"

using namespace rb::podo;

int main() {
  try {
    auto robot = Cobot("10.0.2.7");
    auto rc = ResponseCollector();

    std::array<double, 6> pnt{};

    robot.get_tcp_info(rc, pnt);
    std::cout << "get_tcp_info() : ";
    for(const auto& e : pnt) {
      std::cout << e << " ";
    }
    std::cout << std::endl;

    robot.get_tfc_info(rc, pnt);
    std::cout << "get_tfc_info() : ";
    for(const auto& e : pnt) {
      std::cout << e << " ";
    }
    std::cout << std::endl;

  } catch (const std::exception& e) {
    std::cerr << e.what() << std::endl;
  }

  return 0;
}